src/formatting/expr.rs

```diff
@@ -1612,10 +1616,14 @@ fn rewrite_struct_lit<'a>(
             one_line_width,
         )?
     } else {
-        let field_iter = fields
-            .iter()
-            .map(StructLitField::Regular)
-            .chain(base.into_iter().map(StructLitField::Base));
+        let field_iter = fields.iter().map(StructLitField::Regular).chain(
+            match struct_rest {
+                ast::StructRest::Base(expr) => Some(StructLitField::Base(&**expr)),
+                ast::StructRest::Rest(span) => Some(StructLitField::Rest(span)),
+                ast::StructRest::None => None,
+            }
+            .into_iter(),
+        );
```

Comment:
I couldn't quite convince myself that it'd be worthwhile to try to implement IntoIterator for the StructRest enum, especially given that rustfmt and RLS are currently broken on nightly. This can always be refactored later on if a better solution presents itself